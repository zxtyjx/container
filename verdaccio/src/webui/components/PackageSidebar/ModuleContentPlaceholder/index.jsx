import React from 'react';
import PropTypes from 'prop-types';

import classes from './style.scss';

export default function ModuleContentPlaceholder({text}) {
  return <p className={classes.emptyPlaceholder}>{text}</p>;
}
ModuleContentPlaceholder.propTypes = {
  text: PropTypes.string.isRequired
};
