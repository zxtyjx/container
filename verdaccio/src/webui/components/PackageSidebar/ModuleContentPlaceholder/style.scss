@import '../../../styles/variable';

.emptyPlaceholder {
  text-align: center;
  margin: 20px 0;
  font-size: 16px;
  color: $text-grey;
}
