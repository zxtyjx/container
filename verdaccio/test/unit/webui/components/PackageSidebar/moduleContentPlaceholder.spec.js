/**
 * ModuleContentPlaceholder component
 */

import React from 'react';
import { shallow } from 'enzyme';
import ModuleContentPlaceholder from '../../../../../src/webui/components/PackageSidebar/ModuleContentPlaceholder/index';

console.error = jest.fn();

describe('<PackageSidebar /> : <ModuleContentPlaceholder />', () => {
  it('should error for required props', () => {
    shallow(<ModuleContentPlaceholder />);
    expect(console.error).toBeCalled();
  });
  it('should load module component', () => {
    const props = {
      text: 'Test text'
    };
    const wrapper = shallow(<ModuleContentPlaceholder {...props} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
