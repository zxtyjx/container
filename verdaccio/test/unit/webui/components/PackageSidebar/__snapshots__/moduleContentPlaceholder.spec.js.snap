// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PackageSidebar /> : <ModuleContentPlaceholder /> should load module component 1`] = `"<p class=\\"emptyPlaceholder\\">Test text</p>"`;
